src/assemble_workflow/dist.py

```diff
@@ -16,9 +17,10 @@
 
 
 class Dist(ABC):
-    def __init__(self, name, path):
+    def __init__(self, name, path, distribution):
         self.name = name
         self.path = path
+        self.distribution = distribution
```

Comment:
This shouldn't be necessary since the class `DistRpm`, `DistZip` "knows" what kind of distribution it is. If you need the name, then add a property called `.name` and return it. If you need the extension, add a property called `.extension` and return ".rpm", ".tar.gz", etc.

Reply:
Hi @dblock Thank you so much for your comments! But I'm a little bit confused about how to implement it if we don't add self.distribution to the Dist class. In DistRpm class, we need the distribution of "rpm". Do you mean to simply add a variable called distribution_name = "rpm" in the DistRpm class and use this variable whenever we use self.distribution before?

Reply:
`DistZip` knows it's a zip and `DistRpm` knows it's rpm, so something like this:

```
class DistZip
    @property
    def distribution(self):
        return "zip"
```

Reply:
I see. We only need "rpm" in the DistRpm class